File: kotlin-java/src/main/kotlin/com/fastcampus/kotlinjava/gettersetter/Student.java
package com.fastcampus.kotlinjava.gettersetter;

import java.time.LocalDate;
import java.time.LocalDateTime;

public class Student {

    public String name;
    public LocalDate birthDate;

    public void setName(String name) {
        this.name = name;
    }

    public void setBirthDate(LocalDate birthDate) {
        this.birthDate = birthDate;
    }

    public String getName() {
        return name;
    }

    public LocalDate getBirthDate() {
        return birthDate;
    }
}

File: kotlin-java/src/main/kotlin/com/fastcampus/kotlinjava/gettersetter/Student.kt
package com.fastcampus.kotlinjava.gettersetter

import java.time.LocalDate

class Student {

    @JvmField
    var name: String? = null

    var birthDate: LocalDate? = null
    val age: Int = 10

    var grade: String? = null
        private set //외부에서 setter 생성 불가

    fun changeGrade(grade: String) {
        this.grade = grade
    }
}